### PR TITLE
DOC/UML: Adding UML diagram for UCP

### PR DESCRIPTION
**Pull request by shamisp**
This PR has to go after #451 proposed by @MattBBaker

### CODE REVIEW

**Comment by shamisp** [comment]
@yosefe @miked-mellanox for now please review correctness of the diagram.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1026/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/452/ for details (Mellanox internal link).

File: doc/uml/ucp.dot [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        fontname = "Bitstream Vera Sans"
+        fontsize = 8
+        ]
+
+        ucp_config_t [
+        label = "{ucp_config_t|\l ucp_config_read()\l ucp_config_release()\l ucp_config_print()\l}"
+        ]
+
+        ucp_context_h [
+        label = "{ucp_context_h|\l ucp_init()\l ucp_cleanup\l}"
```
Comment (yosefe) [inline doc/uml/ucp.dot]: need to add () for ucp_cleanup

File: doc/uml/ucp.dot [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        ]
+
+        ucp_worker_h [
+        label = "{ucp_worker_h|\l ucp_worker_create()\l ucp_worker_destroy()\l ucp_worker_progress_register()\l ucp_worker_progress_unregister()\l ucp_worker_get_address()\l ucp_worker_release_address()\l ucp_worker_progress()\l}"
```
Comment (yosefe) [inline doc/uml/ucp.dot]: lets remove progress register/unregister

File: doc/uml/ucp.dot (hunk 1 above)
Comment (shamisp) [inline doc/uml/ucp.dot]: ok

File: doc/uml/ucp.dot (hunk 2 above)
Comment (shamisp) [inline doc/uml/ucp.dot]: good point

**Comment by yosefe** [comment]
:+1:

**Pull request merged**